src/middlewared/middlewared/plugins/rdma/interface/crud.py

```diff
@@ -0,0 +1,251 @@
+                                                       check.get('ping_mac')))
+            if check:
+                self.logger.info(f'Validated communication of {netdev} with IP {check["ping_ip"]}')
+            return True
+        except CheckProcessException:
+            self.logger.info(f'Failed to validate communication of {netdev} with IP {check["ping_ip"]}')
+        except Exception:
+            self.logger.error('Failed to configure RDMA interface', exc_info=True)
+
+        return False
+
+    async def ping(self, node, ifname, ip, mac=None):
+        if not node or node == await self.middleware.call('failover.node'):
+            # Local
+            result = await self.middleware.call('rdma.interface.local_ping', ifname, ip, mac)
+        else:
+            # Remote
+            try:
+                result = await self.middleware.call('failover.call_remote',
+                                                    'rdma.interface.local_ping', [ifname, ip, mac])
+            except CallError as e:
+                if e.errno != CallError.ENOMETHOD:
+                    raise
+                self.logger.warning('Failed to ping from remote RDMA interface')
+                return False
+
+        return result
+
+    async def ifname_to_netdev(self, ifname):
+        links = await self.middleware.call('rdma.get_link_choices')
+        for link in links:
+            if link['rdma'] == ifname:
+                return link['netdev']
+
+    async def local_ping(self, ifname, ip, mac=None):
+        netdev = await self.middleware.call('rdma.interface.ifname_to_netdev', ifname)
+        if not await self.middleware.call('core.ping_remote', {'hostname': ip,
+                                                               'timeout': 1,
+                                                               'count': 4,
+                                                               'interface': netdev,
+                                                               'interval': '0.1'}):
+            # Common case no logging needed
+            return False
+        if mac:
+            macs = await self.middleware.call('core.arp', {'ip': ip, 'interface': netdev})
+            if ip not in macs:
+                # If we can ping it, we should be able to get the MAC address
+                self.middleware.logger.debug('Could not obtain arp info for IP: %s', ip)
+                return False
+            if macs[ip] != mac:
+                # MAC address mismatch
+                self.middleware.logger.debug('MAC address mismatch for IP %s', ip)
+                return False
+        return True
+
+    async def internal_interfaces(self):
+        links = await self.middleware.call('rdma.get_link_choices')
+        ifname_to_netdev = {}
+        for link in links:
+            ifname_to_netdev[link['rdma']] = link['netdev']
+
+        if await self.middleware.call('system.product_type') in ['ENTERPRISE', 'SCALE_ENTERPRISE']:
```

Comment:
This will only ever be `SCALE_ENTERPRISE`

Reply:
Maybe simply `system.is_enterprise`?